front50-api/src/main/java/com/netflix/spinnaker/front50/api/model/pipeline/Pipeline.java

```diff
@@ -21,9 +21,11 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import lombok.EqualsAndHashCode;
 import lombok.Getter;
 import lombok.Setter;
 
+@EqualsAndHashCode
```

Comment:
CAUTION:  This is... going to do a deep comparison on all fields as a result.  It'd be better to do it on explicit fields (aka id) or similar.  I could see this leading to some performance issues as a result.

Reply:
would you think the three fields - id, name and application be better?

Reply:
yeah :) We don't want deep introspection as this would have some potentially nasty impacts elsewhere.  Frankly ID is probably ALL that's needed as that's supposed to be unique, but I'd be ok with name/application as well if we really want.